bittensor/_neuron/text/core_validator/__init__.py

```diff
@@ -669,44 +680,25 @@ def neuron_stats_update(self, neuron_stats: Dict[int, Dict[str, Any]]):
 
         return responsive_uids, list(neuron_stats.keys())  # responsive_uids, queried_uids
 
-    def calculate_weights(self, responsive_uids: Set, queried_uids: Set):
+    def calculate_weights(self):
         r""" Calculates neuron set-weights from weight_key mapped values. Defines weight_key as the neuron stats key
         used to obtain the mapped stat value (typically a Shapley value) that the final set-weights are calculated from.
         """
 
         weight_key = self.weight_key + '!'  # use zeroing key to penalize non-responsive neurons
 
-        # === Randomize UIDs in preferred order (responsive -> queried -> rest) ===
         min_allowed_weights = self.subtensor.min_allowed_weights
         max_weight_limit = self.subtensor.max_weight_limit
 
-        non_responsive_uids = queried_uids - responsive_uids
-        non_queried_uids = set(range(self.metagraph.n)) - queried_uids
-
-        # random.sample(population, k, *, counts=None): Return a k length list of unique elements chosen from
-        # the population sequence or set. Used for random sampling without replacement (so no uid duplicates expected).
-        preferred_uids = (random.sample(list(responsive_uids), len(responsive_uids)) +
-                          random.sample(list(non_responsive_uids), len(non_responsive_uids)) +
-                          random.sample(list(non_queried_uids), len(non_queried_uids)))  # preferred UID random order
-
-        preferred_uids = torch.LongTensor(preferred_uids)
-
         # === Populate neuron weights ===
         neuron_weights = torch.zeros_like(self.metagraph.S)  # allow unevaluated UIDs for min_allowed_weights
-
         for uid in self.neuron_stats:
             if weight_key in self.neuron_stats[uid]:
                 neuron_weights[uid] = torch.tensor([self.neuron_stats[uid][weight_key]])
 
         # === Filter to non-zero weights ===
-        neuron_weights = neuron_weights[preferred_uids]  # rearrange neuron_weights to match preferred_uids order
-        preferred_uids = preferred_uids[neuron_weights > 0]  # filter to non-zero weights
-        neuron_weights = neuron_weights[neuron_weights > 0]  # filter to non-zero weights
-
-        # === Slice weights_to_set UIDs ===
-        weights_to_set = max([min_allowed_weights, len(responsive_uids)])
-        sample_uids = preferred_uids[:weights_to_set]  # slice to weights_to_set
-        sample_weights = neuron_weights[:weights_to_set]  # slice to weights_to_set
+        sample_uids = torch.argwhere(neuron_weights > 0).squeeze(dim=1)  # find uids with non-zero weight
```

Comment:
Now that we have removed the responsive requirement, do you think we should still limit the validator to set weights once it has fully queried all of the network?

Reply:
So the epoch conditions have been changed to set weights exactly every 250 blocks, regardless of whether the network was fully queried. Validators that are unable to query the network fully in 250 blocks will at least have a partial update ready for setting weights, so the on-chain incentive calculations have a steady influx of updated weights from validators.